src/test/ui/match/issue-70972-dyn-trait.rs

```diff
@@ -4,7 +4,8 @@ fn main() {
     let a: &dyn Send = &7u32;
     match a {
         F => panic!(),
-        //~^ ERROR trait objects cannot be used in patterns
+        //~^ ERROR `&dyn Send` cannot be used in patterns
+        //~| ERROR `&dyn Send` cannot be used in patterns
```

Comment:
What's fascinating about this is that the old *error* was gated behind `include_lint_checks` *despite* being a hard error (fatal, even!) and not a lint.

I guess we're calling `to_pat` twice (once for match checking and another time for MIR building)?
And we don't have a way to indicate that an error was already emitted.
Except... is this what `include_lint_checks` is for? Because then I have an idea:
```rust
let msg = format!(...);
if self.include_lint_checks {
    self.tcx().sess.span_err(self.span, &msg);
} else {
    self.tcx().sess.delay_span_bug(self.span, &msg);
}
```

So that way it's guaranteed to be an error but isn't shown if the same logic was run with `include_lint_checks` enabled.